### PR TITLE
JSUI-2832 Move autoselect prevention to the facet level

### PR DESCRIPTION
**Pull request by ThibodeauJF**
SEARCHAPI-4233 will be required for it to work

https://coveord.atlassian.net/browse/JSUI-2832

[Deploy](https://dashboard.heroku.com/pipelines/a3535101-5bbf-4a5b-a909-47fcf8c9f149)

### CODE REVIEW

**Comment by ThibodeauJF** [comment]
@samisayegh  We got to test the feature end to end for the first time recently. For the DynamicFacet, everything about autoselection was working well, for the HierarchicalFacet, I realized that there was some use cases where we couldn't specify preventing autoselection on a specific value.

Talking to UX we realized that maybe for the DynamicFacet as well we want to prevent ALL facet values from being autoselected on interaction, hence asking the SearchAPI to move/add the parameter on the facet level as well.

File: src/ui/DynamicFacet/DynamicFacet.ts [hunk 1]
```diff
@@ -497,6 +498,11 @@ export class DynamicFacet extends Component implements IDynamicFacet {
     this.dynamicFacetQueryController.enableFreezeFacetOrderFlag();
   }
 
+  public enablePreventAutoSelectionFlag() {
```
[the remark this answers is not in the document]
Comment (ThibodeauJF) [reply src/ui/DynamicFacet/DynamicFacet.ts]: yeah looking at it the controller is created when the class is instantiated so those weren't necessary

File: src/controllers/DynamicFacetQueryController.ts [hunk 2]
```diff
@@ -108,10 +115,11 @@ export class DynamicFacetQueryController {
   }
 
   protected get currentValues(): IFacetRequestValue[] {
-    return this.facet.values.allFacetValues.map(({ value, state, preventAutoSelect }) => ({
+    return this.facet.values.allFacetValues.map(({ value, state }) => ({
       value,
       state,
-      preventAutoSelect
+      // TODO: remove after SEARCHAPI-4233 is completed
+      preventAutoSelect: this.preventAutoSelection
```
Comment (olamothe) [inline src/controllers/DynamicFacetQueryController.ts]: What are the consequences if we don't do this ?

Does it just become a "superfluous" parameter, or is there are other consequences ?

Comment (ThibodeauJF) [reply src/controllers/DynamicFacetQueryController.ts]: It becomes a double dip parameter, specifying it both on the facet and all the facet values will have the same behaviour

**Pull request merged**